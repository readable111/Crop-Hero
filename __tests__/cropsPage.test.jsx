/****
 * @author Isaac Boodt
 * @reviewer
 * @tester 
 ***/

import { render } from "@testing-library/react-native";
import CropsPage from "../app/cropspage";
import { useLocalSearchParams } from 'expo-router';
import * as Font from 'expo-font';

jest.mock('expo-font', ()=>({
        useFonts: jest.fn(),
        loadAsync: jest.fn().mockResolvedValue(true),
        isLoaded: jest.fn(()=>true)
}))

const searchParams = { label: 'temp', name: 'temp', hrfNum: '00', active: 'NaN', location: 'Lorem Ipsum', variety: 'Test', source: 'The store', date: '00/00/1001', comments: 'Who knows', indoors: 'Maybe', type:'Weird'};

jest.mock('expo-router', () => ({
        useLocalSearchParams: jest.fn(),
}));

describe('<CropsPage/>', () =>{
        test('renders correctly', () =>{
                useLocalSearchParams.mockReturnValue(searchParams);

                const tree = render(<CropsPage/>).toJSON();
                expect(tree).toMatchSnapshot();
        })
})