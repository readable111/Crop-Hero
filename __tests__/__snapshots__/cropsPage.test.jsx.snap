// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<cropsPage/> renders correctly 1`] = `<cropsPage />`;
